### PR TITLE
Extract Slack notifier from SirCAL to allow reuse

### PR DESCRIPTION
**Pull request by benjaminysmith**
### Description
Extract Slack notifier from SirCAL to allow reuse

### Changelog
Itemize code/test/documentation changes and files added/removed.
- Creates slack notifier
- Refactors SirCAL code to use the extracted code

### CODE REVIEW

File: _delphi_utils_python/delphi_utils/slack_notifier.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    def post_message(self, blocks):
+        """Post a message to the slack channel"""
+        try:
+            self.client.chat_postMessage(
+                channel=self.slack_channel,
+                blocks=blocks
+            )
+        except SlackApiError as e:
+            # You will get a SlackApiError if "ok" is False
+            assert False, e.response["error"]
```
Comment (sgsmob) [inline _delphi_utils_python/delphi_utils/slack_notifier.py:20]: IIUC, this is turning a `SlackApiError` into an `AssertionError` with the same message.  Why not just let the `SlackApiError` go uncaught?

Comment (benjaminysmith) [reply _delphi_utils_python/delphi_utils/slack_notifier.py]: Yeah good point. Removed.

**Review by sgsmob — APPROVED** [review]

**Comment by krivard** [comment]
I think I clobbered you with #675 :(

**Comment by benjaminysmith** [comment]
> I think I clobbered you with #675 :(

Fixed.

**Pull request merged**